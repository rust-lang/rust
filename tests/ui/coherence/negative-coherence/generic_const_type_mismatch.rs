//! This test used to ICE (#119381), because relating the `u8` and `i8` generic
//! const with the array length of the `Self` type was succeeding under the
//! assumption that an error had already been reported.

#![feature(with_negative_coherence)]
trait Trait {}
impl<const N: u8> Trait for [(); N] {}
impl<const N: i8> Trait for [(); N] {}
//~^ ERROR: conflicting implementations of trait `Trait`

fn main() {}
