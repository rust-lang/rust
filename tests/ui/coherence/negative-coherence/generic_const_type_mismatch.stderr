error[E0119]: conflicting implementations of trait `Trait` for type `[(); _]`
  --> $DIR/generic_const_type_mismatch.rs:8:1
   |
LL | impl<const N: u8> Trait for [(); N] {}
   | ----------------------------------- first implementation here
LL | impl<const N: i8> Trait for [(); N] {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `[(); _]`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0119`.
